programs/dual-farming/src/pool.rs

```diff
@@ -0,0 +1,141 @@
+use crate::*;
+use spl_math::uint::U192;
+
+const SECONDS_IN_YEAR: u64 = 365 * 24 * 60 * 60;
+
+pub fn reward_per_token(
+    pool: &Account<Pool>,
+    total_staked: u64,
+    last_time_reward_applicable: u64,
+) -> (u128, u128) {
+    if total_staked == 0 {
+        return (
+            pool.reward_a_per_token_stored,
+            pool.reward_b_per_token_stored,
+        );
+    }
+
+    let time_period = U192::from(last_time_reward_applicable)
+        .checked_sub(pool.last_update_time.into())
+        .unwrap();
+    let a = pool
+        .reward_a_per_token_stored
+        .checked_add(
+            time_period
+                .checked_mul(pool.reward_a_rate.into())
+                .unwrap()
+                .checked_mul(PRECISION.into())
+                .unwrap()
+                .checked_div(SECONDS_IN_YEAR.into())
+                .unwrap()
+                .checked_div(total_staked.into())
+                .unwrap()
+                .try_into()
+                .unwrap(), //back to u128
+        )
+        .unwrap();
+
+    let b = pool
+        .reward_b_per_token_stored
+        .checked_add(
+            time_period
+                .checked_mul(pool.reward_b_rate.into())
+                .unwrap()
+                .checked_mul(PRECISION.into())
+                .unwrap()
+                .checked_div(SECONDS_IN_YEAR.into())
+                .unwrap()
+                .checked_div(total_staked.into())
+                .unwrap()
+                .try_into()
+                .unwrap(), //back to u128
+        )
+        .unwrap();
+
+    (a, b)
+}
+
+pub fn rate_after_funding(
+    pool: &mut Account<Pool>,
+    _reward_a_vault: &Account<TokenAccount>,
```

Comment:
Let's remove `_reward_a_vault` and `_reward_b_vault` since we are not using the `RewardCalculator` trait anymore

Reply:
Fixed